icu4c/source/i18n/formattedval_sbimpl.cpp

```diff
@@ -219,12 +219,19 @@ bool FormattedValueStringBuilderImpl::nextPositionImpl(ConstrainedFieldPosition&
     }
 
     U_ASSERT(currField == kUndefinedField);
+    // Always set the position to the end so that we don't revisit previous sections
+    cfpos.setState(
+        cfpos.getCategory(),
+        cfpos.getField(),
+        fString.fLength,
+        fString.fLength);
     return false;
 }
 
 void FormattedValueStringBuilderImpl::appendSpanInfo(int32_t spanValue, int32_t length) {
-    if (spanIndices.getCapacity() <= spanValue) {
-        spanIndices.resize(spanValue * 2);
+    U_ASSERT(spanIndices.getCapacity() >= spanValue);
```

Comment:
U_ASSERT( A >= B) follow by a if ( A == B) does not look right to me. You should change the assert to U_ASSERT(spanIndices.getCapacity() >= spanValue) I think . If it is always >=  as the invariant of the code then the if == won't happen.

Reply:
I believe it better be 
U_ASSERT(spanIndices.getCapacity() > spanValue);
if (spanIndices.getCapacity() <= spanValue) {